extra/src/main/java/aima/extra/probability/ProbabilityNumber.java

```diff
@@ -0,0 +1,149 @@
+package aima.extra.probability;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+
+/**
+ * ProbabilityNumber interface defines arithmetic specifically for probability
+ * values. A probability value is quantified between 0 and 1. Operations that
+ * violate the property of a probability value are not permitted. This interface
+ * is implemented by the various ProbabilityNumber implementations.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public interface ProbabilityNumber extends Comparable<ProbabilityNumber> {
+
+	/**
+	 * Get the probability value represented by the ProbabilityNumber
+	 * implementation as a standard BigDecimal.
+	 * 
+	 * @return value of BigDecimal type.
+	 */
+	BigDecimal getValue();
+
+	/**
+	 * Get the MathContext object associated with the ProbabilityNumber object.
+	 * MathContext can be set for BigDecimalProbabilityNumber and
+	 * RationalProbabilityNumber via the constructor mechanism. The default
+	 * MathContext object associated with the implementations is of DECIMAL128
+	 * precision. It is possible to go upto an even higher precision all the way
```

Comment:
the -> these. Not sure the defaults for these implementations need to be mentioned here but ok.